### ISSUE TITLE
data linked {{for start= end=}} renders all items on $.observable.refresh()

### ISSUE THREAD

**Issue by johan-ohrn** [BorisMoore/jsviews#410]
I managed to reduce the problem to a short [jsfiddle](https://jsfiddle.net/17yoefvs/1/)

Run the fiddle with the browser console open and you will see that the initial render only renders 10 items.
After a second the timeout does an observable refresh on the array. At this time hundreds of renders take place and the 10 items I want rendered are even rendered twice. Most renders are just discarded. This is a major performance bottle neck for us right now.
Any ideas?

For future reference the html is:

```
<div class="container">
</div>

```
And javascript is:

```
var data = {blah: []};
for (var i = 0; i < 1000; i++) {
 data.blah.push(i);
}
$.views.templates("mytemplate", "\
  {^{for blah start=900 end=910 }}\
    {{:~log('render: '+#data)}}<br>\
  {{/for}}\
").link('.container', data, {
	log: function(x) {
  	console.log(x);
    return x;
  }
})

setTimeout(function() {
 $.observable(data.blah).refresh(data.blah.slice().reverse())
},1000);
```

**Comment by BorisMoore** [comment]
Thanks for calling this out, and providing simple repro... I'll look into it...

**Comment by johan-ohrn** [comment]
I found out that if you use refresh with the same array instance then the extra renders don't happen.

The call to .slice() creates a new copy of the array and then .reverse() reverses the copy
```$.observable(data.blah).refresh(data.blah.slice().reverse())```

Without the call to .slide() .reverse() performs an in-place reverse on the same array instance that is data linked
```$.observable(data.blah).refresh(data.blah.reverse())```

It's a workaround but it's subtle and we're likely to end up doing what I'm doing in the first example. Especially if the sorting is something more complicated than reverse.

**Comment by BorisMoore** [comment]
Removing the slice() means that you are first unobservably reversing the blah array then calling refresh with its now current (reversed) items

    data.blah.reverse();
    $.observable(data.blah).refresh(data.blah);

So while it appears to work in the jsfiddle example, if you try instead doing the same but with no start/end settings:

    {^{for blah}}

then the array change (reverse) will not trigger updated rendering at all.

I have a tentative fix for your issue. I may test it a bit more, but if you are interested you could try it, and let me know if it does indeed resolve your issues:

It's a one line fix on line 3975

    // if (!changing) {
    if (!changing && !eventArgs.refresh) {

[jsviews.js.txt](https://github.com/BorisMoore/jsviews/files/2636625/jsviews.js.txt)

**Comment by johan-ohrn** [comment]
I just tried the fix in our code base and it works perfectly!
Thanks alot!

**Comment by BorisMoore** [comment]
Fixed in commit v1.0.1
